libs/execution_context/ebpf_maps.c

```diff
@@ -439,6 +445,311 @@ _next_hash_map_key(_In_ ebpf_core_map_t* map, _In_ const uint8_t* previous_key,
     return result;
 }
 
+static ebpf_core_map_t*
+_create_hash_map_per_cpu(_In_ const ebpf_map_definition_t* map_definition)
+{
+    ebpf_result_t retval;
+    size_t map_size = sizeof(ebpf_core_map_t);
+    ebpf_core_map_t* map = NULL;
+    uint32_t cpu_count;
+    uint32_t index;
+    size_t per_cpu_size = 0;
+    ebpf_core_map_per_cpu_t* per_cpu = NULL;
+
+    ebpf_get_cpu_count(&cpu_count);
```

Comment:
can cpu_count ever change at runtime?

Reply:
Yes. Windows supports hot-add of CPUs.

Reply:
Note: when we find the per-cpu state, we check if the current CPU <= allocated CPU count, so if a hot add occurs we won't store state for the new CPU until the program is reloaded.

Reply:
that's what I was going to ask.  What does linux do?